### ISSUE TITLE
Sporadic failed connections to server

### ISSUE THREAD

**Issue by nhammond** [StanfordBioinformatics/loom#60]
We sometimes see the error below in the daemon log on the server.

This appears to occur when using sqlite but not with mysql.

Often this results in a steprun being finished, with files uploaded, but with its status never set to "complete".
```
ERROR [2016-04-29 18:52:32,272] ('Connection broken: IncompleteRead(0 bytes read
)', IncompleteRead(0 bytes read))
Traceback (most recent call last):
  File "/opt/loom/lib/python2.7/site-packages/loom/master/analysis/management/co
mmands/run_tasks.py", line 22, in handle
    response = requests.get(settings.MASTER_URL_FOR_WORKER+'/api/controls/run/')
  File "/opt/loom/local/lib/python2.7/site-packages/requests/api.py", line 67, i
n get
    return request('get', url, params=params, **kwargs)
  File "/opt/loom/local/lib/python2.7/site-packages/requests/api.py", line 53, i
n request
    return session.request(method=method, url=url, **kwargs)
  File "/opt/loom/local/lib/python2.7/site-packages/requests/sessions.py", line 
468, in request
    resp = self.send(prep, **send_kwargs)
  File "/opt/loom/local/lib/python2.7/site-packages/requests/sessions.py", line 
608, in send
    r.content
  File "/opt/loom/local/lib/python2.7/site-packages/requests/models.py", line 73
7, in content
    self._content = bytes().join(self.iter_content(CONTENT_CHUNK_SIZE)) or bytes
()
  File "/opt/loom/local/lib/python2.7/site-packages/requests/models.py", line 66
3, in generate
    raise ChunkedEncodingError(e)
ChunkedEncodingError: ('Connection broken: IncompleteRead(0 bytes read)', Incomp
leteRead(0 bytes read))
```

**Comment by isaacliao** [comment]
Switched to MySQL using Google Cloud SQL and this still happens.

I noticed that it's always in run_tasks.py (never update_tasks.py or any other request). Maybe sometimes it takes too long to call run() on all the TaskRuns?

**Comment by nhammond** [comment]
I've also seen this several times today on a Loom server with a local mysql database.

I noticed there were 4 TaskRuns when only 3 were expected. 2 of the 4 were incomplete. I have not determined if this occurs with all of the failures of this type, as I haven't been checking for this.

**Comment by nhammond** [comment]
Fixed in 3876e2c133fad62d752ec2207ca868a83c22a7b8

Added retry mechanism to client connections. Loom recovers from failed http connections, e.g. if the database is locked.
